Azure DevOps enhancements

kmontocam opened the pull request:
Replaced `AZURE_ICON` because of mdi, appended proper `VCS_GIT_AZURE_ICON` using DevOps icon

romkatv commented:
What problem are you trying to solve?

kmontocam commented:
The current nerdfont icons are depreciated or inadequate.
<img width="864" alt="image" src="https://github.com/romkatv/powerlevel10k/assets/80984024/a03131d8-dc22-4cd9-a899-650e56227c69">

Both default Azure Icon and VCS have newer versions, I modified the icons for the "nerdfont-complete" catalog.

kmontocam commented:
Although the Azure DevOps icon was listed, there was no case to match that this was being used as the remote server

romkatv commented:
Is something not working for you? If so, please describe the issue in the following terms:

1. What you do.
2. What you expect to happen.
3. What actually happens.

kmontocam commented:
Working with the Azure Cloud, both with the CLI tools and DevOps have outdated/missing features.

First of all, p10k detects if the user is typing an Azure CLI command. When it does, it shows the Azure subscription Name, followed by an Azure Icon. The icon included in the current version of p10k for Azure is depreciated on the current nerdfont standard.

Currently (nf-mdi-azure)
`
AZURE_ICON                     '\uFD03'               # ﴃ
`

Proposed changes (nf-cod-azure)
`
AZURE_ICON                     '\uEBD8'               # 
`


The same can be said for the DevOps icons. There is a more adequate one that can be replaced. Still on `'nerdfont-complete'|'nerdfont-fontconfig'`, we have:

Currently (nf-mdi-azure)
`
VCS_GIT_AZURE_ICON             '\uFD03 '              # ﴃ
`

Proposed changes:
`
VCS_GIT_AZURE_ICON             '\uEBE8 '              # 
`

Finally, there is not a case to match that the git remote server that the user is working with is on Azure DevOps thru an HTTP method. This was only capable of matching SSH connections. The expression to match this case was modified like in the other Git Clouds.

romkatv commented:
What do the http remotes on Azure look like? Can you give an example?

What is the problem with the icon? Does it not display for you? If not, please run `p10k configure` and let me know if it helps.

kmontocam commented:
> What do the http remotes on Azure look like? Can you give an example?
> 
> What is the problem with the icon? Does it not display for you? If not, please run `p10k configure` and let me know if it helps.

The HTTP remotes look with the default GIT icon instead of the Azure DevOps:

Example of DevOps (sorry I can not show the complete repo URL, i will censor a part of it):
![Untitled](https://github.com/romkatv/powerlevel10k/assets/80984024/47991b54-a535-4b79-be3e-45fb7b98539b)
Side Note. Is it normal that the stages don't have a space next to the number? I saw that these are not given an extra space like
```
    icons[VCS_BOOKMARK_ICON]+=' '
    icons[VCS_COMMIT_ICON]+=' '
    icons[VCS_BRANCH_ICON]+=' '
```

Working with the Azure Client. The icon does display, but as this _Uyghur?_ character. Example:
![Untitled](https://github.com/romkatv/powerlevel10k/assets/80984024/7a9eec72-1778-4140-a4f7-bc0245c7b504)

kmontocam commented:
> Thanks for the details.
> 
> Does the icon get fixed if you run `p10k configure`?

No. If I use, MesloLGS NF I can see the old icon typing `az`, but with other Nerd Fonts I see ﴃ because they have depreciated that icon. And for Azure DevOps git issue, I still see the default git icon, not the DevOps one

romkatv commented:
Could you post screenshots of the wizard screens related to icons along with your answers?

kmontocam commented:
Okay, I got you. I had a config with MelosNF which is a nerdfont-complete and I was using a nerdfont-v3. Now it all makes sense.

> Could you post screenshots of the wizard screens related to icons along with your answers?

Okay, I got you. I had a `p10k config` with MelosNF which is a nerdfont-complete and I was using a nerdfont-v3. Now it all makes sense. I set it up again (with the font I use, which happens to be a v3) and now I see the old azure icon when I type `az`. That is also why I did not see the spacing between the stashed and the numbers. Sorry for that.

However, I still think it will be good to update those icons at least in nerdfont-v3, as well as applying the changes for the Azure DevOps git indicator, because after the new config, I still see the default git icon because is not detecting the https remote server.

romkatv commented:
I'll update the pattern for the azure remote.

It's not yet clear to me whether there is an issue with the icon. Do you have a font with which the icon does not render after you run `p10k configure` and truthfully answer all questions?

kmontocam commented:
> I'll update the pattern for the azure remote.
> 
> It's not yet clear to me whether there is an issue with the icon. Do you have a font with which the icon does not render after you run `p10k configure` and truthfully answer all questions?

Yes, but it was my bad because I had a p10k configuration answering with yes all question, but then I switched to a font that didn't meet that requirement (had a no in the upper arrow). So yeah, the only "issue" right now with the icons is that the Azure DevOps one is the same as the Azure Icon, and there is an specialized one for DevOps (code: ebe8 in nerdfonts), and there a new one for Azure icon (code: ebd8) because of the change of logo in Azure.

In the fork I introduced the new logos for "nerdfont-v3", as well as the changes for the pattern for the azure remote

romkatv commented:
I've updated the remote URL pattern for Azure devops and changed Azure and Azure DevOps icons when using NerdFonts v3. Thanks for bringing these issues to my attention.

In the future please file an issue instead of sending a PR. It'll save us both time.
